stdlib/gettext.pyi

```diff
@@ -27,6 +27,15 @@ class GNUTranslations(NullTranslations):
     CONTEXT: str
     VERSIONS: Sequence[int]
 
+@overload
+def find(
+    domain: str, localedir: StrPath | None = ..., languages: Iterable[str] | None = ..., all: Literal[True] = ...
+) -> list[str]: ...
+@overload
+def find(
+    domain: str, localedir: StrPath | None = ..., languages: Iterable[str] | None = ..., all: Literal[False] = ...
+) -> str | None: ...
+@overload
 def find(domain: str, localedir: StrPath | None = ..., languages: Iterable[str] | None = ..., all: bool = ...) -> Any: ...
```

Comment:
```suggestion
    domain: str, localedir: StrPath | None = ..., languages: Iterable[str] | None = ..., all: Literal[True]
) -> list[str]: ...
@overload
def find(
    domain: str, localedir: StrPath | None = ..., languages: Iterable[str] | None = ..., all: Literal[False] = ...
) -> str | None: ...
@overload
def find(domain: str, localedir: StrPath | None = ..., languages: Iterable[str] | None = ..., all: bool) -> Any: ...
```

I think this should get rid of the "overlapping overloads" complaint!

Reply:
I think we can't do that. Because in your example `all` without default follows args with defaults. It is a syntax error.

I've tested this piece to work correctly with a single ignore:

```python
from typing import Any, overload, Literal

@overload  # ignores incompatible overloads
def find(  # type: ignore[misc]
    domain: str, all: Literal[False] = ...
) -> str | None: ...
@overload
def find(
    domain: str, all: Literal[True] = ...
) -> list[str]: ...
@overload
def find(
    domain: str, all: bool = ...
) -> Any: ...

reveal_type(find('a'))              # N: Revealed type is "Union[builtins.str, None]"
reveal_type(find('a', all=True))    # N: Revealed type is "builtins.list[builtins.str]"
reveal_type(find('a', all=False))   # N: Revealed type is "Union[builtins.str, None]"
reveal_type(find('a', all=1 in {})) # N: Revealed type is "Any"
```

Btw, `Any` here still looks a bit to wide to me 🤔

Reply:
Oh shoot, I see.

> Btw, Any here still looks a bit to wide to me 🤔

Maybe `str | None | Any` (or something similarly 🌟fancy🌟)?

Reply:
The usual solution is to split each non-`= ...` overload into two cases: one that has the argument (and everything before it) specified positionally, and another where it is specified as a keyword argument. This would be a total of 5 overloads, but it would prevent overlapping overloads. I'm fine with it either way, but if you think that would be cleaner, do that instead :)

Reply:
I think that `3` with a `type: ignore` comment is cleaner 👍